src/components/info-window.tsx

```diff
@@ -19,38 +25,76 @@ export const InfoWindow = (props: PropsWithChildren<InfoWindowProps>) => {
   const {children, anchor, onCloseClick, ...infoWindowOptions} = props;
   const map = useContext(GoogleMapsContext)?.map;
 
+  const infoWindow = useRef<google.maps.InfoWindow | null>(null);
   const [contentContainer, setContentContainer] =
     useState<HTMLDivElement | null>(null);
 
   // create infowindow once map is available
   useEffect(() => {
     if (!map) return;
 
-    const infoWindow = new google.maps.InfoWindow(infoWindowOptions);
+    const newInfowindow = new google.maps.InfoWindow(infoWindowOptions);
 
     // Add content to info window
     const el = document.createElement('div');
-    infoWindow.setContent(el);
-    infoWindow.open({map, anchor});
-
-    if (onCloseClick) {
-      google.maps.event.addListener(infoWindow, 'closeclick', () => {
-        onCloseClick();
-      });
-    }
+    newInfowindow.setContent(el);
 
+    infoWindow.current = newInfowindow;
     setContentContainer(el);
 
     // Cleanup info window and event listeners on unmount
     return () => {
-      google.maps.event.clearInstanceListeners(infoWindow);
+      google.maps.event.clearInstanceListeners(newInfowindow);
 
-      infoWindow.close();
+      newInfowindow.close();
       el.remove();
 
       setContentContainer(null);
     };
-  }, [map, children, anchor]);
+    // We don't want to re-render a whole new infowindow
+    // when the options change to prevent flickering.
+    // Update of infoWindow options is handled in the useEffect below.
+    // Excluding infoWindowOptions from dependency array on purpose here.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [map, children]);
+
+  // Update infoWindowOptions
+  useEffect(() => {
```

Comment:
this may or may not cause problems: due to destructuring (line 25) `infoWindowOptions` will be considered changed for every render. In #181 we'll add a [`useDeepCompareEffect`](https://github.com/visgl/react-google-maps/pull/181/files#diff-5118501a9135d733bb840b77dcafcfaf23f121e4eae24be4d630ef1e968a1dc2) that we can use here instead.

I'd say leave it like this for now.